### ISSUE TITLE
aria-controls test fails if id is inside shadow DOM

### ISSUE THREAD

**Issue by YonatanKra** [dequelabs/axe-core#4044]
### Product

axe-core

### Product Version

4.7.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
Describe what you expected the product to do.

#### Actual
I have the following setup:
```
<custom-element aria-controls="id">
  #shadowRoot
    <div id="id"></div>
</custom-element>
```
This fails for this reason:
`Invalid ARIA attribute value: aria-controls="id"`

Here's an example from our `select` component:
![image](https://github.com/dequelabs/axe-core/assets/6459899/a46e5dc7-856c-4ebe-978f-6e0eb366dfca)
`vwc-select` has `aria-controls` and the selected div has `id="listbox-0"`.

#### How to Reproduce
https://codepen.io/yonatankra/pen/mdzGRxx

**Comment by WilcoFiers** [comment]
That's intentional @YonatanKra. `aria-controls` is what's called an IDREF property, where you reference an element by its ID. ID referencing only works when both elements are in the same (shadow) DOM tree. That's by design. Shadow DOM trees are meant to be encapsulating. Nothing outside that tree should effect how that tree functions. That includes IDs. One of the major benefits of that is that it means you don't have to worry about potential duplicate IDs that may exist in other trees. You can repeat a duplicate a shadow DOM tree without accidentally having incorrect ID references. The downside is that there's no way to reference elements that are in a different tree.

I'm afraid the only option is to rearchitect that component so the listbox isn't in a different tree. (Also, I'm afraid there's a good bit more wrong with this component going off that screenshot. role=select is not a valid role. And your `label` element references a `div`, which does not work either.)

**Comment by YonatanKra** [comment]
Thanks @WilcoFiers .

We are aware of the issues and already fixed them (the screenshot is an old one).

It was just that `aria-controls` issue was found via the node API and is not showing when using the chrome extension (nor lighthouse).

Thanks again for the detailed response :)

**Comment by WilcoFiers** [comment]
Don't think there's any further action on this. Closing the issue.
